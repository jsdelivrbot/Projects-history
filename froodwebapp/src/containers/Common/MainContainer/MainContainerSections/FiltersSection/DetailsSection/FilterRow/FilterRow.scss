.filterRow {
  margin-top: 1rem;
}