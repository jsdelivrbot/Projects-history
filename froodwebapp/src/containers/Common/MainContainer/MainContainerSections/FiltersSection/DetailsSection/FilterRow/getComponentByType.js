/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { FroodInput, FroodDatePicker } from 'components';

const getComponentByType = (
  id,
  type,
  value,
  handleFilterInputChange,
  handleDatePickerChange
) => {
  switch (true) {
    case type.includes('datetime'):
      return (
        <FroodDatePicker
          id={ id.toString() }
          onChange={ handleDatePickerChange }
          value={ value }
        />
      );
    case type.includes('varchar'):
      return (
        <FroodInput
          id={ id }
          value={ value }
          onChange={ handleFilterInputChange }
        />);
    default:
      return null;
  }
};

export default getComponentByType;
