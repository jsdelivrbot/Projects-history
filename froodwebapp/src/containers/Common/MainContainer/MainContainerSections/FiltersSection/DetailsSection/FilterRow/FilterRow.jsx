/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-flexbox-grid';
import { FroodSelect } from 'components';
import { AddIcon, DeleteIcon } from 'icons';
import { logicOperators, relationalOperators, getMenuItems } from 'utils';
import getComponentByType from './getComponentByType';

const FilterRow = ({
  rowIndex,
  rowData: {
    id,
    value,
    columnId,
    logicalOpId,
    relationalOpId
  },
  filterColumns,
  handleChangeColumn,
  handleChangeColumnAvailableValue,
  handleChangeLogicalOperator,
  handleChangeRelationalOperator,
  handleFilterInputChange,
  handleDatePickerChange,
  handleAddFilterRow,
  handleDeleteFilterRow
}) => {

  const {
    tableDataType,
    availableValues
  } = filterColumns.find(col => col.id === Number(columnId));

  let relationalOperatorsItems = relationalOperators;

  if (tableDataType.includes('varchar')) {
    relationalOperatorsItems = relationalOperators.filter(op => op.symbol === '=' || op.symbol === '!=');
  }

  return (
    <Row middle="xs">
      <Col xs={ 2 } sm={ 1 } md={ 1 } lg={ 1 }>
        { rowIndex !== 0 &&
          <FroodSelect
            id={ id }
            menuItems={ logicOperators }
            value={ logicalOpId }
            onChange={ handleChangeLogicalOperator }
          />
        }
      </Col>
      <Col xs sm={ 3 } md={ 3 } lg={ 2 }>
        <FroodSelect
          id={ id }
          menuItems={ getMenuItems(filterColumns) }
          value={ columnId }
          onChange={ handleChangeColumn }
        />
      </Col>
      <Col xs sm={ 2 } md={ 2 } lg={ 2 }>
        <FroodSelect
          id={ id }
          menuItems={ relationalOperatorsItems }
          value={ relationalOpId }
          onChange={ handleChangeRelationalOperator }
        />
      </Col>
      { availableValues &&
        <Col xs sm={ 4 } md={ 4 } lg={ 3 }>
          <FroodSelect
            id={ id }
            value={ value }
            menuItems={ availableValues }
            onChange={ handleChangeColumnAvailableValue }
          />
        </Col>
      }
      { !availableValues &&
        <Col xs sm={ 4 } md={ 4 } lg={ 3 }>
          { getComponentByType(
            id, // filterId
            tableDataType.toLowerCase(), // columnDataType
            value, // filterValue
            handleFilterInputChange, // Input Change handler
            handleDatePickerChange // DatePicker Change handler
          )}
        </Col>
      }
      <Col xs sm={ 2 } md={ 2 } lg={ 1 }>
        <AddIcon
          id={ id }
          onClick={ handleAddFilterRow }
        />
        { rowIndex !== 0 &&
          <DeleteIcon
            id={ id }
            onClick={ handleDeleteFilterRow }
          />
        }
      </Col>
    </Row>
  );
};

FilterRow.propTypes = {
  filterColumns: PropTypes.array.isRequired,
  rowIndex: PropTypes.number.isRequired,
  rowData: PropTypes.object.isRequired,
  handleChangeColumn: PropTypes.func.isRequired,
  handleChangeColumnAvailableValue: PropTypes.func.isRequired,
  handleChangeLogicalOperator: PropTypes.func.isRequired,
  handleChangeRelationalOperator: PropTypes.func.isRequired,
  handleFilterInputChange: PropTypes.func.isRequired,
  handleDatePickerChange: PropTypes.func.isRequired,
  handleAddFilterRow: PropTypes.func.isRequired,
  handleDeleteFilterRow: PropTypes.func.isRequired
};

export default FilterRow;

