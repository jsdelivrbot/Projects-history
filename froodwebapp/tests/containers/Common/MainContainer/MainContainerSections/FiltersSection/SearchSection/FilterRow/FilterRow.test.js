import FilterRow from 'containers/Common/MainContainer/MainContainerSections/FiltersSection/SearchSection/FilterRow/FilterRow';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('FilterRow', () => {
  const props = {
    filterColumns: [{
      id: 1,
      order: 'first order',
      displayName: 'Inventory',
      isDefault: false,
      tableDataType: 'Table data of first column'
    }, {
      id: 2,
      order: 'second order',
      displayName: 'Price lists',
      isDefault: true,
      tableDataType: 'Table data of second column'
    }],
    rowIndex: 1,
    rowData: {
      id: 1,
      value: 'Some value',
      columnId: 1,
      logicalOpId: 1,
      relationalOpId: 1
    },
    handleFilterChange: jest.fn(),
    handleFilterInputChange: jest.fn(),
    handleFilterNameChange: jest.fn(),
    handleDatePickerChange: jest.fn(),
    handleAddFilterRow: jest.fn(),
    handleDeleteFilterRow: jest.fn(),
    handleSaveFilter: jest.fn(),
    handleDeleteFilter: jest.fn(),
    handleChangeColumn: jest.fn(),
    handleChangeColumnAvailableValue: jest.fn(),
    handleChangeLogicalOperator: jest.fn(),
    handleChangeRelationalOperator: jest.fn(),
  };

  it('renders the FilterRow component', () => {
    const component = shallow(<FilterRow { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of FilterRow', () => {
    const component = shallow(<FilterRow { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });
});
