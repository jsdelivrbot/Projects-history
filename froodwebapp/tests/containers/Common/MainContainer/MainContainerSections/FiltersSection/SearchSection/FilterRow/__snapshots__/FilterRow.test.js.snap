// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterRow capturing Snapshot of FilterRow 1`] = `
<Row
  middle="xs"
>
  <Col
    lg={1}
    md={1}
    sm={1}
    xs={2}
  >
    <FroodSelect
      id={1}
      menuItems={
        Array [
          Object {
            "key": 7,
            "symbol": "and",
            "type": "2",
            "value": "And",
          },
          Object {
            "key": 8,
            "symbol": "or",
            "type": "2",
            "value": "Or",
          },
        ]
      }
      onChange={[Function]}
      value={1}
    />
  </Col>
  <Col
    lg={2}
    md={3}
    sm={3}
    xs={true}
  >
    <FroodSelect
      id={1}
      menuItems={
        Array [
          Object {
            "key": 1,
            "value": undefined,
          },
          Object {
            "key": 2,
            "value": undefined,
          },
        ]
      }
      onChange={[Function]}
      value={1}
    />
  </Col>
  <Col
    lg={2}
    md={2}
    sm={2}
    xs={true}
  >
    <FroodSelect
      id={1}
      menuItems={
        Array [
          Object {
            "key": 1,
            "symbol": "=",
            "type": "1",
            "value": "Is equal to",
          },
          Object {
            "key": 2,
            "symbol": ">",
            "type": "1",
            "value": "Is greater than",
          },
          Object {
            "key": 3,
            "symbol": "<",
            "type": "1",
            "value": "Is less than",
          },
          Object {
            "key": 4,
            "symbol": ">=",
            "type": "1",
            "value": "Is greater than equal to",
          },
          Object {
            "key": 5,
            "symbol": "<=",
            "type": "1",
            "value": "Is less than equal to",
          },
          Object {
            "key": 6,
            "symbol": "!=",
            "type": "1",
            "value": "Not equal to",
          },
        ]
      }
      onChange={[Function]}
      value={1}
    />
  </Col>
  <Col
    lg={3}
    md={4}
    sm={4}
    xs={true}
  >
    <div />
  </Col>
  <Col
    lg={1}
    md={2}
    sm={2}
    xs={true}
  />
</Row>
`;
